bench/lib/Experiments.hs

```diff
@@ -401,11 +431,14 @@ setup = do
 --------------------------------------------------------------------------------------------
 
 -- Parse the max residency in RTS -s output
-parseMaxResidency :: String -> Int
-parseMaxResidency input =
-  case find ("maximum residency" `isInfixOf`) (reverse $ lines input) of
-    Just l -> read $ filter isDigit $ head (words l)
-    Nothing -> -1
+parseMaxResidencyAndAllocations :: String -> (Int, Int)
+parseMaxResidencyAndAllocations input =
+    (f "maximum residency", f "bytes allocated in the heap")
+  where
+    inps = reverse $ lines input
+    f label = case find (label `isInfixOf`) inps of
+        Just l -> read $ filter isDigit $ head (words l)
```

Comment:
Fixing bracket and $ is a bit odd.

```suggestion
        Just l -> read $ filter isDigit $ head $ words l
```

Or you might want to use `word1` from Extra.